pkg/frontend/transport/seeker_body.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package transport
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// ParseSeekerBodyForm will ensure that request body is a Seeker, parse the form and seek the body back to the beginning.
+func ParseSeekerBodyForm(r *http.Request) error {
+	return withSeekerBody(r, func(r *http.Request) error {
+		return r.ParseForm()
+	})
+}
+
+// withSeekerBody will ensure that request body is a Seeker, call the provided function and seek the body back to the beginning.
+func withSeekerBody(r *http.Request, f func(r *http.Request) error) (err error) {
```

Comment:
Arguably resetting body back to beginning is the most important part of this function. Name should reflect that.